internal/command/jsonformat/computed/renderers/map.go

```diff
@@ -14,31 +14,44 @@ import (
 var _ computed.DiffRenderer = (*mapRenderer)(nil)
 
 func Map(elements map[string]computed.Diff) computed.DiffRenderer {
-	maximumKeyLen := 0
-	for key := range elements {
-		if maximumKeyLen < len(key) {
-			maximumKeyLen = len(key)
-		}
+	return &mapRenderer{
+		elements: elements,
 	}
+}
 
+func NestedMap(elements map[string]computed.Diff) computed.DiffRenderer {
 	return &mapRenderer{
-		elements:      elements,
-		maximumKeyLen: maximumKeyLen,
+		elements:                  elements,
+		overrideNullSuffix:        true,
+		overrideForcesReplacement: true,
 	}
 }
 
 type mapRenderer struct {
 	NoWarningsRenderer
 
-	elements      map[string]computed.Diff
-	maximumKeyLen int
+	elements map[string]computed.Diff
+
+	overrideNullSuffix        bool
+	overrideForcesReplacement bool
 }
 
 func (renderer mapRenderer) RenderHuman(diff computed.Diff, indent int, opts computed.RenderHumanOpts) string {
+	forcesReplacementSelf := diff.Replace && !renderer.overrideForcesReplacement
+	forcesReplacementChildren := diff.Replace && renderer.overrideForcesReplacement
+
 	if len(renderer.elements) == 0 {
-		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(diff.Replace, opts.OverrideForcesReplacement))
+		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(forcesReplacementSelf, opts.OverrideForcesReplacement))
 	}
 
+	maximumKeyLen := 0
+	for key := range renderer.elements {
+		if maximumKeyLen < len(key) {
+			maximumKeyLen = len(key)
```

Comment:
Sorry, my brain is fogging up after all this reviewing, but shouldn't this be using `ensureValidAttributeName` like we do with the object renderer?

Reply:
We always escape map keys unlike object keys, so I was handling that part elsewhere. But I think it makes sense to do all the escaping in the same place. So I introduced a new function that does the escaping, `ensureValueAttributeName` calls it and the map renderer calls it directly for every key.